File: Java/pbPlots/Example1.java
package pbPlots;

import static pbPlots.pbPlots.*;
import static pbPlots.SupportLib.*;

public class Example1{
	public static void main(String [] args){
		boolean success;
		StringReference errorMessage = new StringReference();
		RGBABitmapImageReference imageReference = CreateRGBABitmapImageReference();

		double [] xs = {-2, -1, 0, 1, 2};
		double [] ys = {2, -1, -2, -1, 2};

		success = DrawScatterPlot(imageReference, 600, 400, xs, ys, errorMessage);

		if(success) {
			double[] pngdata = ConvertToPNG(imageReference.image);
			WriteToFile(pngdata, "example1.png");
			DeleteImage(imageReference.image);
		}else{
			System.err.println(new String(errorMessage.string));
		}
	}
}

File: VisualBasic/example1.vb
Module Example1
	Sub Main(args As String())
		Dim imageReference As RGBABitmapImageReference = CreateRGBABitmapImageReference()

		Dim xs as double() = {-2, -1, 0, 1, 2}
		Dim ys as double() = {2, -1, -2, -1, 2}

		Dim errorMessage As StringReference = New StringReference()
		Dim success As Boolean = DrawScatterPlot(imageReference, 800, 600, xs, ys, errorMessage)

        If success
            Dim pngdata as double() = ConvertToPNG(imageReference.image)
            Call WriteToFile(pngdata, "example1.png")
            Call DeleteImage(imageReference.image)
        Else
            Console.Error.WriteLine(errorMessage.stringx)
        End If
	End Sub
End Module
